### DISCUSSION TITLE
VM Interfaces permissions missing after upgrade

### DISCUSSION THREAD

**Discussion by cslingerland** [netbox-community/netbox#6944] [Q&A] [answered]
We migrated to a new server with the following steps, and everything is working fine except for one thing. The permissions for virtualization.xxxx_vminterface (introduced in 2.9) are missing. I created them manually through the admin interfaces (I was able to select "Virtualization > Interfaces" when creating a new permission), but for some reason they were not there by default. I fear that there may be others that I'm missing, and I'm unsure why this happened, or how to correct it.

1. exported database from server running 2.7.10, following [official netbox instructions](https://netbox.readthedocs.io/en/stable/administration/replicating-netbox/).
2. imported database to new server running 2.11.10
3. manually ran migrations on the new server to update the database

Like I said, everything else is working, but people weren't able to see any VM interfaces (except the superusers) until I created the missing permissions manually and assigned them to appropriate parties.

**Comment by sdktr** [comment, marked as answer]
The models that are added in a newer version are not automatically added to permissions.
Since VM interface is a new entity this is expected behavior. For this scenario, where the model is actually split of from some other model it could have dupped the permissions as well I guess, but I don't believe this was added.

**Reply by sdktr** [reply]
https://github.com/netbox-community/netbox/issues/4721

Here is that change

**Reply by cslingerland** [reply]
Thank you for that. So if it's expected behavior, are you saying that any new permission that gets added in the future won't show up? If so, then at what point are permissions created? During initial install of Netbox? 
Is there a way to get them to show up? It seems like something that could introduce a lot of headaches to system admins if permissions that are referenced in code don't get added to the choices.

**Reply by ryanmerolle** [reply]
@sdktr is just saying the model, and there was no migration to see if permission had the old model set to then go and set the new model in the permission.

Each jump in version has its release notes which explain what is needed or what you should be aware of.
